torch/quantization/ns/pattern_utils.py

```diff
@@ -65,9 +69,23 @@ def get_reversed_fusions() -> Set[Tuple[NSFusionType, int]]:
     all_quant_patterns = get_default_quant_patterns()
     default_base_op_idx = 0
     for quant_pattern, _quant_handler in all_quant_patterns.items():
-        # this only takes patterns of multiple ops
+        # Only patterns of multiple ops are fusions, ignore
+        # patterns which contain a single ops (they get matched
+        # without caring about fusions).
         if isinstance(quant_pattern, tuple):
-            results.add((quant_pattern, default_base_op_idx))  # type: ignore[arg-type]
+            results.append((quant_pattern, default_base_op_idx))  # type: ignore[arg-type]
+
+        # For each pattern, add additional patterns with observers and
+        # fake quants at the end.
+        # TODO(future PR): if needed, implement matching for a node
+        #   having multiple output observers.
+        for cls in (ObserverBase, FakeQuantizeBase):
```

Comment:
Would there be the cases where both input and output of the op have observers or fake_quant attached, or observers and fake_quant only appear at the op output? My impression is for QAT we attach fake_quant to both input and output?

Reply:
yeah I think so, is the current pattern defined as fusion_pattern + fake_quantize/observer at the output?

Reply:
here is a way to think about it

```
# fp32
x0 -> conv0 -> x1

# prepared
x0 -> fq_of_x0 -> conv0 -> fq_of_conv0 -> x1

# prepared, with input and output loggers
x0 -> fq_of_x0 -> input_logger_of_conv0 -> conv0 -> fq_of_conv0 -> output_logger_of_conv0 -> x1
```

in this way of thinking about it, the input of the previous op is a black box and it shouldn't matter whether it has observers or not.  We can insert input and output loggers around the current op and its observer, and it should work as needed.

Reply:
Thanks for the explanation! It makes more sense now. 

One case I'm thinking is in QAT we could add fake_quant to both input and output of the op, and want to insert logger around them, for exmaple:
```
# prepared 
x0 -> fq_in_of_conv0 -> conv0 -> fq_out_of_conv0 -> x1

# prepared, with input and output loggers 
x0 ->  input_logger_of_conv0->fq_in_of_conv0 -> conv0 -> fq_out_of_conv0->output_logger_of_conv0 -> x1
```
is it possible to make this function more flexible to handle above case?